assignments/arrays.js

```diff
@@ -58,39 +58,94 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // arr = [1,2,3,4];
 // for (i = 0; i < arr.length; i++) {
-//     arr[i]; // 1,2,3,4
+//	 arr[i]; // 1,2,3,4
 // }
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
+function challenge1(arr){
+	for(let i=0; i<arr.length; i++){
+		if(arr[i].id==33) return arr[i];
+	}
+}
+
+let car = challenge1(inventory);
+
+// or with array methods
+// let car = inventory.filter((elem) => elem.id==33)[0];
+
+console.log(`Car 33 is a ${car.car_year} ${car.car_make} ${car.car_model}` );
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+function challenge2(arr){
```

Comment:
The last car they were looking for is {"id":50,"car_make":"Lincoln","car_model":"Town Car","car_year":1999} based on the array. Your solution for challenge 2 returns the wrong one

Reply:
Aww, I misunderstood it somehow & overcomplicated this one as well - don't know why, but I thought it's all about the newest car out there. Fixed in #2.

```js
function challenge2(arr){
    return arr[arr.length-1];
}
```